category-service/src/main/java/com/edmart/category/serviceImpl/CategoryServiceImpl.java

```diff
@@ -21,51 +26,84 @@ public class CategoryServiceImpl implements CategoryService {
 
     private final CategoryRepository categoryRepository;
 
+    private final CategoryMapper categoryMapper;
+
+
     @Override
-    public void createCategory(CategoryDTO categoryDTO) {
-        try {
-            categoryRepository.save(CategoryMapper.INSTANCE.mapToCategory(categoryDTO));
+    public void createCategory(CategoryDTO categoryDTO) throws CategoryNotFoundException {
+        Category category = new Category();
+        try{
+            category.setCategoryName(categoryDTO.categoryName());
+            category.setCategoryDescription(categoryDTO.categoryDescription());
+            category.setCategoryDesignation(categoryDTO.categoryDesignation());
 
-        }catch (Exception e){
-            log.error("Error creating category with name {} caused by {}",categoryDTO.categoryName(),e.getCause());
+            categoryRepository.save(category);
+        }catch(Exception e){
+            log.error("Error Creating category with name {} caused by {}", categoryDTO.categoryName(), e.getCause());
         }
     }
 
     @Override
-    public List<CategoryDTO> getAllCategories() {
-        return categoryRepository.findAll().stream()
-                .map(CategoryMapper.INSTANCE::mapToCategoryDTO)
-                .collect(Collectors.toList());    }
+    public CategoryResponseDTO getAllCategories(int page, int size,String sortBy, String sortDir) {
+        Sort sort = sortDir.equalsIgnoreCase(Sort.Direction.DESC.name())
+                ? Sort.by(sortBy).ascending()
+                : Sort.by(sortBy).descending();
+
+        Pageable pageable = PageRequest.of(page, size, sort);
+
+        Page<Category> categoryPage = categoryRepository.findAll(pageable);
+
+        CategoryResponseDTO categoryResponseDto=new CategoryResponseDTO();
+        categoryResponseDto.setCategoryList(categoryPage.getContent());
+        categoryResponseDto.setPageInfoDTO(Pagination.getPageInfoDTO(categoryPage));
+
+        return  categoryResponseDto;
+    }
 
     @Override
-    public CategoryDTO getCategory(Long categoryId) {
-        return categoryRepository.findById(categoryId)
-                .map(CategoryMapper.INSTANCE::mapToCategoryDTO)
-                .orElseThrow(() -> new CategoryNotFoundException("Category not found"));
+    public CategoryDTO getCategory(Long categoryId) throws CategoryNotFoundException {
+        Optional<CategoryDTO> categoryDTOOptional = categoryRepository.findById(categoryId)
+                .map(categoryMapper);
+        if(categoryDTOOptional.isPresent()){
```

Comment:
This is great but you could also use the Optional ifPresentOrElse method if you are not an if.....else lover. This method takes a consumer and an action as parameters.

categoryDTOOptional.ifPresentOrElse(
                foundCategory -> {
                    return foundCategory;
                },
                () -> {
                    Throw new CategoryNotFoundException("Category with this Id does not exist!")     
           }
        );